src/password.rs

```diff
@@ -44,26 +28,12 @@ impl FromStr for SafePassword {
     type Err = PasswordError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Ok(Self::from(s.to_owned()))
-    }
-}
-
-impl SafePassword {
-    /// Gets a reference to bytes of a passphrase.
-    pub fn reveal(&self) -> &[u8] {
-        self.password.reveal()
+        Ok(Self::hide(<&str as Into<String>>::into(s).into_bytes()))
     }
 }
 
```

Comment:
We can keep the test and a new one to verify that the values are masked, properly.

```suggestion

#[cfg(test)]
mod tests {
    use crate::SafePassword;
    use std::any::type_name;

    use super::SafePasswordLabel;

    #[test]
    fn test_password() {
        assert_eq!(
            SafePassword::from("secret_must_match".to_string()),
            SafePassword::from("secret_must_match")
        );
    }

    #[test]
    fn it_masks() {
        let password = SafePassword::from("my_secret_is_mine");
        let formatted_password = format!("{}", password);
        let expected_format = format!(
            "Hidden<{}, {}>",
            type_name::<Vec<u8>>(),
            type_name::<SafePasswordLabel>()
        );
        assert_eq!(formatted_password, expected_format);
    }
}
```

Reply:
Yeah, I plan to add more comprehensive tests if the design is accepted.